OpenSSL/SSL.py

```diff
@@ -1883,6 +1883,18 @@ def get_cipher_version(self):
             return version.decode("utf-8")
 
 
+    def get_protocol_version(self):
+        """
+        Obtain the protocol version of the current connection.
+
+        :returns: The TLS version of the current connection, for example
+            the value for TLS 1.2 would be 0x303.
+        :rtype: :py:class:`unicode`
+        """
+        version = _ffi.string(_lib.SSL_get_version(self._ssl))
+        return version.decode("utf-8")
```

Comment:
Dunno, the name of the protocol version definitely seems textual to me.

On Mon, Apr 27, 2015 at 10:16 AM, Hynek Schlawack <notifications@github.com>
wrote:

> In OpenSSL/SSL.py
> <https://github.com/pyca/pyopenssl/pull/244#discussion_r29149506>:
>
> > @@ -1883,6 +1883,18 @@ def get_cipher_version(self):
> >              return version.decode("utf-8")
> >
> >
> > +    def get_protocol_version(self):
> > +        """
> > +        Obtain the protocol version of the current connection.
> > +
> > +        :returns: The TLS version of the current connection, for example
> > +            the value for TLS 1.2 would be 0x303.
> > +        :rtype: :py:class:`unicode`
> > +        """
> > +        version = _ffi.string(_lib.SSL_get_version(self._ssl))
> > +        return version.decode("utf-8")
>
> I’m not sure about this one. I believe we decided to keep pyOpenSSL’s APIs
> bytes-based except for paths.
>
> So unless someone can enlighten me on what I’m missing, I’m gonna claim
> this ought to be bytes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/pyopenssl/pull/244/files#r29149506>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084